packages/backend/src/modules/tvl2/services/ValueService.ts

```diff
@@ -0,0 +1,97 @@
+import { assert } from '@l2beat/backend-tools'
+
+import { AmountConfigEntry, ProjectId, UnixTime } from '@l2beat/shared-pure'
+import { AmountRepository } from '../repositories/AmountRepository'
+import { PriceRepository } from '../repositories/PriceRepository'
+import { ValueRecord } from '../repositories/ValueRepository'
+import { calculateValue } from '../utils/calculateValue'
+import { AmountId } from '../utils/createAmountId'
+import { AssetId, createAssetId } from '../utils/createAssetId'
+import { PriceId } from '../utils/createPriceId'
+
+interface Values {
+  canonical: bigint
+  canonicalForTotal: bigint
+  external: bigint
+  externalForTotal: bigint
+  native: bigint
+  nativeForTotal: bigint
+}
+
+export interface ValueServiceDependencies {
+  priceRepository: PriceRepository
+  amountRepository: AmountRepository
+}
+
+export class ValueService {
+  constructor(private readonly $: ValueServiceDependencies) {}
+
+  async getTvlAt(
```

Comment:
I'm not a fan of this name, as the `at` suffix was indicating that it's happening for a single timestamp.

Reply:
yeah, good catch, it has not been renamed after abstracting it to the service